backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -374,6 +369,56 @@ export class ExperimentAssignmentService {
         condition: monitoredDocument.condition || logDocument.condition,
       };
     } else {
+      // experiment level exclusion
```

Comment:
Before this change, the enrollment exclusions were only updated when the experiment state was 'enrolling' or 'enrollment complete'. But this could run when the experiment is in any state. Is that a concern?
 
Also, I noticed that almost the identical code runs at the end of each branch of the `if` that test whether there are experiments after experiment-level exclusions are applied (line 256). 
However, in the first there is
``     const logDocument = await this.monitoredDecisionPointLogRepository.save({
        monitoredDecisionPoint: monitoredDocument,
        condition: condition,
        uniquifier: uniquifier,
      });
``
while in the second theres
``      const logDocument = await this.monitoredDecisionPointLogRepository.save({
        monitoredDecisionPoint: monitoredDocument,
        condition: assignmentUnit === ASSIGNMENT_UNIT.WITHIN_SUBJECTS ? condition : null,
        uniquifier: assignmentUnit === ASSIGNMENT_UNIT.WITHIN_SUBJECTS ? uniquifier : null,
      });
``
Do we only want to test for WITHIN_SUBJECTS to set the 'condition' and 'uniquifier' to `null` in the case where there's experiment-level exclusion?